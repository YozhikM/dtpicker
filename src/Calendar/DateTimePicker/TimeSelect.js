/* @flow */

import React from 'react';
import SvgIcon from '../../SvgIcon/SvgIcon';
import InputUpDown from '../InputUpDown/InputUpDown';
import TableSelect, { type Options } from '../TableSelect/TableSelect';
import './TimeSelect.scss';

type Value = Date;

type ShowEnum = 'main' | 'hh' | 'mm' | 'ss';

type HHMMSS = {
  hh: number,
  mm: number,
  ss: number,
};

type Props = {|
  value?: ?Value,
  onSetTime?: (value: Value) => void,
  onSubmit?: (value: Value) => void,
  showSeconds?: boolean,
  show?: ShowEnum,
|};

type State = HHMMSS & {
  show: ShowEnum,
};

class TimeSelect extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      ...this.parseTime(this.props.value),
      show: this.props.show || 'main',
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    if (this.props.show !== nextProps.show) {
      this.setState({ show: nextProps.show });
    }
  }

  genTableOpts = (from: number, till: number, step: number = 1): Options => {
    const res: Options = [];
    for (let i = from; i <= till; i += step) {
      res.push({ value: i, name: i < 10 ? `0${i}` : `${i}` });
    }
    return res;
  };

  parseTime(value: ?Date): HHMMSS {
    if (!value) return { hh: 0, mm: 0, ss: 0 };
    return { hh: value.getHours(), mm: value.getMinutes(), ss: value.getSeconds() };
  }

  getValue = (): Value => {
    const { hh, mm, ss } = this.state;
    const { value } = this.props;
    const newDate = value ? new Date(value.getTime()) : new Date();
    newDate.setHours(hh, mm, ss);
    return newDate;
  };

  onSubmit = () => {
    const { onSubmit } = this.props;
    if (onSubmit) onSubmit(this.getValue());
  };

  onSetTime = () => {
    const { onSetTime } = this.props;
    if (onSetTime) onSetTime(this.getValue());
  };

  onChangeHH = (hh: number) => {
    this.setState({ hh }, this.onSetTime);
  };

  onChangeMM = (mm: number) => {
    this.setState({ mm }, this.onSetTime);
  };

  onChangeSS = (ss: number) => {
    this.setState({ ss }, this.onSetTime);
  };

  onClickHH = () => {
    this.setState({ show: 'hh' });
  };

  onClickMM = () => {
    this.setState({ show: 'mm' });
  };

  onClickSS = () => {
    this.setState({ show: 'ss' });
  };

  onChangeTableHH = (value: number) => {
    this.setState({ hh: value, show: 'main' }, this.onSetTime);
  };

  onChangeTableMM = (value: number) => {
    this.setState({ mm: value, show: 'main' }, this.onSetTime);
  };

  onChangeTableSS = (value: number) => {
    this.setState({ ss: value, show: 'main' }, this.onSetTime);
  };

  render() {
    const { showSeconds } = this.props;
    const { hh, mm, ss, show } = this.state;

    if (show === 'hh') {
      return (
        <TableSelect
          options={this.genTableOpts(0, 23)}
          cols={6}
          value={hh}
          onChange={this.onChangeTableHH}
        />
      );
    } else if (show === 'mm') {
      return (
        <TableSelect
          options={this.genTableOpts(0, 59, 5)}
          cols={4}
          value={mm}
          onChange={this.onChangeTableMM}
        />
      );
    } else if (show === 'ss' && showSeconds) {
      return (
        <TableSelect
          options={this.genTableOpts(0, 59, 5)}
          cols={6}
          value={ss}
          onChange={this.onChangeTableSS}
        />
      );
    }

    return (
      <div className="hs">
        <InputUpDown
          value={hh}
          min={0}
          max={23}
          onChange={this.onChangeHH}
          onClickNumber={this.onClickHH}
          circular
        />
        <InputUpDown
          value={mm}
          min={0}
          max={59}
          onChange={this.onChangeMM}
          onClickNumber={this.onClickMM}
          circular
        />
        {showSeconds && (
          <InputUpDown
            value={ss}
            min={0}
            max={59}
            onChange={this.onChangeSS}
            onClickNumber={this.onClickSS}
            circular
          />
        )}
        <button onClick={this.onSubmit}>
          <SvgIcon file="check" />
        </button>
      </div>
    );
  }
}

export default TimeSelect;
