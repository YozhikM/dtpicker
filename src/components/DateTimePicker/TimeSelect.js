/* @flow */

import React from 'react';
import InputUpDown from '../InputUpDown/InputUpDown';
import SvgIcon from '../SvgIcon';
import TableSelect, { type Options } from '../TableSelect/TableSelect';
import s from './TimeSelect.scss';

type Value = Date;

type HHMMSS = {
  hh: number,
  mm: number,
  ss: number
};

type Props = {
  value?: Value,
  onChange?: (value: Value) => void,
  onSubmit?: () => void,
  showSeconds?: boolean
};

type State = HHMMSS & {
  show: string
};

function genTableOpts(from: number, till: number, step: number = 1): Options {
  const res: Options = [];
  for (let i = from; i <= till; i = i + step) {
    res.push({ value: i, name: i < 10 ? `0${i}` : `${i}` });
  }
  return res;
}

class TimeSelect extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      ...this.parseTime(this.props.value),
      show: 'main'
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    if (this.props.value !== nextProps.value) {
      this.setState(this.parseTime(nextProps.value));
    }
  }

  parseTime(value: ?Date): HHMMSS {
    if (!value) return { hh: 0, mm: 0, ss: 0 };
    return { hh: value.getHours(), mm: value.getMinutes(), ss: value.getSeconds() };
  }

  getValue(): Value {
    const { hh, mm, ss } = this.state;
    const value = this.props.value || new Date();
    value.setHours(hh);
    value.setMinutes(mm);
    value.setSeconds(ss);
    return value;
  }

  onChange = () => {
    const { onChange } = this.props;
    if (onChange) {
      onChange(this.getValue());
    }
  };

  onSubmit = () => {
    const { onSubmit } = this.props;
    if (onSubmit) {
      onSubmit();
    }
  };

  onChangeHH = (hh: number) => {
    this.setState({ hh }, this.onChange);
  };

  onChangeMM = (mm: number) => {
    this.setState({ mm }, this.onChange);
  };

  onChangeSS = (ss: number) => {
    this.setState({ ss }, this.onChange);
  };

  onClickHH = () => {
    this.setState({ show: 'hh' });
  };

  onClickMM = () => {
    this.setState({ show: 'mm' });
  };

  onClickSS = () => {
    this.setState({ show: 'ss' });
  };

  render() {
    const { showSeconds } = this.props;
    const { hh, mm, ss, show } = this.state;

    if (show === 'hh') {
      return (
        <TableSelect
          options={genTableOpts(0, 23)}
          cols={6}
          value={hh}
          onChange={value => {
            this.setState({ hh: value, show: 'main' }, this.onChange);
          }}
        />
      );
    } else if (show === 'mm') {
      return (
        <TableSelect
          options={genTableOpts(0, 59, 5)}
          cols={4}
          value={mm}
          onChange={value => {
            this.setState({ mm: value, show: 'main' }, this.onChange);
          }}
        />
      );
    } else if (show === 'ss') {
      return (
        <TableSelect
          options={genTableOpts(0, 59, 5)}
          cols={6}
          value={ss}
          onChange={value => {
            this.setState({ ss: value, show: 'main' }, this.onChange);
          }}
        />
      );
    }

    return (
      <div className={s.hs}>
        <InputUpDown
          value={hh}
          min={0}
          max={23}
          onChange={this.onChangeHH}
          onClickNumber={this.onClickHH}
          circular
        />
        <InputUpDown
          value={mm}
          min={0}
          max={59}
          onChange={this.onChangeMM}
          onClickNumber={this.onClickMM}
          circular
        />
        {showSeconds && (
          <InputUpDown
            value={ss}
            min={0}
            max={59}
            onChange={this.onChangeSS}
            onClickNumber={this.onClickSS}
            circular
          />
        )}
        <button onClick={this.onSubmit}>
          <SvgIcon file="check" />
        </button>
      </div>
    );
  }
}

export default TimeSelect;
