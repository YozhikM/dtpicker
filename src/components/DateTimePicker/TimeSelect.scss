.hs {
  display: flex;
  width: 600px;
  justify-content: center;

  button {
    border: none;
    background-color: transparent;
    outline: none;
    cursor: pointer;
    margin: 0 !important;
    padding: 0 !important;
  }

  p {
    display: block;
    width: 100%;
    text-align: center;
    padding: 5%;
    border: 2px solid rgb(52, 73, 94);
    cursor: pointer;
  }
}
