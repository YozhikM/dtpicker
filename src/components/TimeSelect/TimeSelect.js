/* @flow */

import React from 'react';
import InputUpDown from '../InputUpDown/InputUpDown';
import SvgIcon from '../SvgIcon';
import TableSelect, { type Options } from '../TableSelect/TableSelect';
import s from './TimeSelect.scss';

type Value = Date;

type ShowEnum = 'main' | 'hh' | 'mm' | 'ss';

type HHMMSS = {
  hh: number,
  mm: number,
  ss: number
};

type Props = {
  value?: ?Value,
  onChange?: (value: Value) => void,
  onSubmit?: (value: Value) => void,
  showSeconds?: boolean,
  show?: ShowEnum,
};

type State = HHMMSS & {
  show: ShowEnum,
};

class TimeSelect extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      ...this.parseTime(this.props.value),
      show: this.props.show || 'main',
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    if (this.props.value !== nextProps.value) {
      this.setState(this.parseTime(nextProps.value));
    }
    if (this.props.show !== nextProps.show) {
      this.setState({ show: nextProps.show });
    }
  }

  genTableOpts = (from: number, till: number, step: number = 1): Options => {
    const res: Options = [];
    for (let i = from; i <= till; i) {
      res.push({ value: i, name: i < 10 ? `0${i}` : `${i}` });
    }
    return res;
  };

  parseTime(value: ?Date): HHMMSS {
    if (!value) return { hh: 0, mm: 0, ss: 0 };
    return { hh: value.getHours(), mm: value.getMinutes(), ss: value.getSeconds() };
  }

  getValue = (): Value => {
    const { hh, mm, ss } = this.state;
    let value = this.props.value || new Date();
    value.setHours(hh);
    value.setMinutes(mm);
    value.setSeconds(ss);
    return value;
  };

  onSubmit = () => {
    const { onSubmit } = this.props;
    if (onSubmit) onSubmit(this.getValue());
  };

  onChange = () => {
    const { onChange } = this.props;
    if (onChange) onChange(this.getValue());
  };

  onChangeHH = (hh: number) => {
    this.setState({ hh }, this.onChange);
  };

  onChangeMM = (mm: number) => {
    this.setState({ mm }, this.onChange);
  };

  onChangeSS = (ss: number) => {
    this.setState({ ss }, this.onChange);
  };

  onClickHH = () => {
    this.setState({ show: 'hh' });
  };

  onClickMM = () => {
    this.setState({ show: 'mm' });
  };

  onClickSS = () => {
    this.setState({ show: 'ss' });
  };

  onChangeTableHH = (value: number) => {
    this.setState({ hh: value, show: 'main' }, this.onChange)
  };

  onChangeTableMM = (value: number) => {
    this.setState({ mm: value, show: 'main' }, this.onChange)
  };

  onChangeTableSS = (value: number) => {
    this.setState({ ss: value, show: 'main' }, this.onChange)
  };

  render() {
    const { showSeconds } = this.props;
    const { hh, mm, ss, show } = this.state;

    if (show === 'hh') {
      return (
        <TableSelect
          options={this.genTableOpts(0, 23)}
          cols={6}
          value={hh}
          onChange={this.onChangeTableHH}
        />
      );
    } else if (show === 'mm') {
      return (
        <TableSelect
          options={this.genTableOpts(0, 59, 5)}
          cols={4}
          value={mm}
          onChange={this.onChangeTableMM}
        />
      );
    } else if (show === 'ss' && showSeconds) {
      return (
        <TableSelect
          options={this.genTableOpts(0, 59, 5)}
          cols={6}
          value={ss}
          onChange={this.onChangeTableSS}
        />
      );
    }

    return (
      <div className={s.hs}>
        <InputUpDown
          value={hh}
          min={0}
          max={23}
          onChange={this.onChangeHH}
          onClickNumber={this.onClickHH}
          circular
        />
        <InputUpDown
          value={mm}
          min={0}
          max={59}
          onChange={this.onChangeMM}
          onClickNumber={this.onClickMM}
          circular
        />
        {showSeconds && (
          <InputUpDown
            value={ss}
            min={0}
            max={59}
            onChange={this.onChangeSS}
            onClickNumber={this.onClickSS}
            circular
          />
        )}
        <button onClick={this.onSubmit}>
          <SvgIcon file="check" />
        </button>
      </div>
    );
  }
}

export default TimeSelect;
