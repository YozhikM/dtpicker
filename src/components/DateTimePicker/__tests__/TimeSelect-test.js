/* @flow */

import React from 'react';
import renderer from 'react-test-renderer';
import { setHours, setMinutes, setSeconds } from 'date-fns';
import TimeSelect from '../TimeSelect';

function findBtn(wrapper) {
  return wrapper.find('button').at(4);
}

function findTable(wrapper) {
  return wrapper.find('TableSelect');
}

describe('TimeSelect', () => {
  describe('render()', () => {
    it('from props', () => {
      expect(
        renderer.create(<TimeSelect value={new Date(2017, 8, 8)} showSeconds={true} />)
      ).toMatchSnapshot();
    });

    it('from state', () => {
      let element = <TimeSelect value={new Date(2017, 8, 8)} showSeconds={true} />;
      const wrapper = shallow(element);
      wrapper.setState({ value: new Date(2000, 1, 9) });
      expect(renderer.create(wrapper)).toMatchSnapshot();
    });
  });

  describe('onChange()', () => {
    let value = new Date(2017, 7, 7, 0, 0, 0);
    const onChange = jest.fn();

    beforeEach(() => {
      onChange.mockReset();
    });

    describe('change input', () => {
      describe('change HH', () => {
        it('change state', () => {
          const wrapper = shallow(<TimeSelect value={value} />);
          wrapper.instance().onChangeHH(14);
          expect(wrapper.state('hh')).toBe(14);
        });

        it('onChange callback', () => {
          const wrapper = shallow(<TimeSelect value={value} onChange={onChange} />);
          wrapper.instance().onChangeHH(22);
          expect(onChange).toBeCalledWith(setHours(value, 22));
        });
      });

      describe('change MM', () => {
        it('change state', () => {
          const wrapper = shallow(<TimeSelect value={value} />);
          wrapper.instance().onChangeMM(12);
          expect(wrapper.state('mm')).toBe(12);
        });

        it('onChange callback', () => {
          const wrapper = shallow(<TimeSelect value={value} onChange={onChange} />);
          wrapper.instance().onChangeMM(12);
          expect(onChange).toBeCalledWith(setMinutes(value, 12));
        });
      });

      describe('change SS', () => {
        it('change state', () => {
          const wrapper = shallow(<TimeSelect value={value} showSeconds />);
          wrapper.instance().onChangeSS(7);
          expect(wrapper.state('ss')).toBe(7);
        });

        it('onChange callback', () => {
          const wrapper = shallow(<TimeSelect value={value} onChange={onChange} showSeconds />);
          wrapper.instance().onChangeSS(7);
          expect(onChange).toBeCalledWith(setSeconds(value, 7));
        });
      });
    });
  });

  describe('onClick()', () => {
    const wrapper = shallow(<TimeSelect />);

    describe('click HH', () => {
      it('change state', () => {
        wrapper.instance().onClickHH();
        expect(wrapper.state('show')).toBe('hh');
      });
    });

    describe('click MM', () => {
      it('change state', () => {
        wrapper.instance().onClickMM();
        expect(wrapper.state('show')).toBe('mm');
      });
    });

    describe('click SS', () => {
      it('change state', () => {
        wrapper.instance().onClickSS();
        expect(wrapper.state('show')).toBe('ss');
      });
    });
  });

  it('componentWillReceiveProps()', () => {
    const wrapper = shallow(<TimeSelect showSeconds={true} />);
    wrapper.setProps({ showSeconds: false });
    expect(wrapper.state('showSeconds')).toBeFalsy();
  });

  it('onSubmit()', () => {
    const [wrapper, spy] = spyMount(<TimeSelect />, 'onSubmit');
    findBtn(wrapper).simulate('click');
    findBtn(wrapper);
    expect(spy).toBeCalled();
  });

  it('onChange()', () => {
    const [wrapper, spy] = spyMount(<TimeSelect />, 'onChange');
    wrapper.instance().onChange();
    expect(spy).toBeCalled();
  });

  describe('parseTime()', () => {
    const wrapper = shallow(<TimeSelect showSeconds={true} />);
    let value = new Date(2017, 7, 7, 10, 9, 8);
    let result = wrapper.instance().parseTime(value);

    const { hh, mm, ss } = result;
    expect(hh).toBe(10);
    expect(mm).toBe(9);
    expect(ss).toBe(8);
  });

  describe('getValue() of', () => {
    const wrapper = shallow(<TimeSelect value={new Date(2010, 9, 8, 7, 6, 5)} />);

    it('hours', () => {
      let hh = wrapper.state('hh');
      expect(hh).toBe(7);
    });

    it('minutes', () => {
      let mm = wrapper.state('mm');
      expect(mm).toBe(6);
    });

    it('seconds', () => {
      let ss = wrapper.state('ss');
      expect(ss).toBe(5);
    });
  });

  describe('genTableOpts() of', () => {
    const wrapper = shallow(<TimeSelect value={new Date(2010, 9, 8)} />);

    it('hours table', () => {
      let array = wrapper.instance().genTableOpts(0, 23);
      expect(array).toHaveLength(24);
    });

    it('minutes table', () => {
      let array = wrapper.instance().genTableOpts(0, 59, 5);
      expect(array).toHaveLength(12);
    });

    it('seconds table', () => {
      let array = wrapper.instance().genTableOpts(0, 59, 5);
      expect(array).toHaveLength(12);
    });
  });

  describe('user interactions', () => {
    it('button onClick', () => {
      const onSubmit = jest.fn();
      const [wrapper, spy] = spyMount(
        <TimeSelect value={new Date(2010, 9, 8)} onSubmit={onSubmit} />,
        'onSubmit'
      );
      findBtn(wrapper).simulate('click');
      expect(spy).toBeCalled();
    });

    describe('tables onChange', () => {
      const onChange = jest.fn();
      let wrapper;
      let spy;

      beforeEach(() => {
        [wrapper, spy] = spyMount(
          <TimeSelect value={new Date(2010, 9, 8)} onChange={onChange} />,
          'onChange'
        );
      });

      describe('hh table', () => {
        it('change show', () => {
          wrapper.setState({ show: 'hh' });
          findTable(wrapper)
            .props()
            .onChange();
          expect(wrapper.state('show')).toBe('main');
        });

        it('change state', () => {
          let value = 12;
          wrapper.setState({ show: 'hh' });
          findTable(wrapper)
            .props()
            .onChange(value);
          expect(wrapper.state('hh')).toBe(12);
        });

        it('onChange callback', () => {
          wrapper.setState({ show: 'hh' });
          findTable(wrapper)
            .props()
            .onChange();
          expect(spy).toBeCalled();
        });
      });

      describe('mm table', () => {
        it('change show', () => {
          wrapper.setState({ show: 'mm' });
          findTable(wrapper)
            .props()
            .onChange();
          expect(wrapper.state('show')).toBe('main');
        });

        it('change state', () => {
          let value = 10;
          wrapper.setState({ show: 'mm' });
          findTable(wrapper)
            .props()
            .onChange(value);
          expect(wrapper.state('mm')).toBe(10);
        });

        it('onChange callback', () => {
          wrapper.setState({ show: 'mm' });
          findTable(wrapper)
            .props()
            .onChange();
          expect(spy).toBeCalled();
        });
      });

      describe('ss table', () => {
        it('change show', () => {
          wrapper.setState({ show: 'ss' });
          findTable(wrapper)
            .props()
            .onChange();
          expect(wrapper.state('show')).toBe('main');
        });

        it('change state', () => {
          let value = 9;
          wrapper.setState({ show: 'ss' });
          findTable(wrapper)
            .props()
            .onChange(value);
          expect(wrapper.state('ss')).toBe(9);
        });

        it('onChange callback', () => {
          wrapper.setState({ show: 'ss' });
          findTable(wrapper)
            .props()
            .onChange();
          expect(spy).toHaveBeenCalled();
        });
      });
    });
  });
});
