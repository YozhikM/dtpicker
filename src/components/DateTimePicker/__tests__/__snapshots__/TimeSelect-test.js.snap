// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSelect render() from props 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-up.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
    <p
      onClick={[Function]}
    >
      0
    </p>
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-down.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
  </div>
  <div
    className={undefined}
  >
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-up.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
    <p
      onClick={[Function]}
    >
      0
    </p>
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-down.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
  </div>
  <div
    className={undefined}
  >
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-up.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
    <p
      onClick={[Function]}
    >
      0
    </p>
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-down.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
  </div>
  <button
    onClick={[Function]}
  >
    <div
      className={undefined}
      dangerouslySetInnerHTML={
        Object {
          "__html": "check.svg",
        }
      }
      style={
        Object {
          "display": "inline-block",
          "fill": undefined,
          "height": "24px",
          "width": "24px",
        }
      }
    />
  </button>
</div>
`;

exports[`TimeSelect render() from state 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-up.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
    <p
      onClick={[Function]}
    >
      0
    </p>
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-down.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
  </div>
  <div
    className={undefined}
  >
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-up.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
    <p
      onClick={[Function]}
    >
      0
    </p>
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-down.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
  </div>
  <div
    className={undefined}
  >
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-up.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
    <p
      onClick={[Function]}
    >
      0
    </p>
    <button
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className={undefined}
        dangerouslySetInnerHTML={
          Object {
            "__html": "chevron-down.svg",
          }
        }
        style={
          Object {
            "display": "inline-block",
            "fill": undefined,
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </button>
  </div>
  <button
    onClick={[Function]}
  >
    <div
      className={undefined}
      dangerouslySetInnerHTML={
        Object {
          "__html": "check.svg",
        }
      }
      style={
        Object {
          "display": "inline-block",
          "fill": undefined,
          "height": "24px",
          "width": "24px",
        }
      }
    />
  </button>
</div>
`;
